Automate Release Notes (#109)
nautobot/nautobot-app-dev-example: docs/dev/release_checklist.md

> You will need to have the project's poetry environment built at this stage, as the towncrier command runs **locally only**. If you don't have it, run `poetry install` first.

> Generate release notes with `invoke generate-release-notes --version 1.4.2`. This will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes. If you're releasing a new major or minor version, update the `Release Overview` in `docs/admin/release_notes/version_{major}.{minor}.md` and double check the `mkdocs.yml` release notes navigation.

Reviewer: Since the script is handling the mkdocs.yml updates should it say this instead:

```suggestion
Generate release notes with `invoke generate-release-notes --version 1.4.2`. This will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes. If you're releasing a new major or minor version, update the `Release Overview` in `docs/admin/release_notes/version_{major}.{minor}.md` and `git add mkdocs.yml` to stage any changes made to the release notes navigation.
```

> Generate release notes with `invoke generate-release-notes --version 1.4.2`. This will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes. If you're releasing a new major or minor version, update the `Release Overview` in `docs/admin/release_notes/version_{major}.{minor}.md` and `git add mkdocs.yml` to stage any changes made to the release notes navigation.

Reviewer: ```suggestion
Generate release notes with `invoke generate-release-notes --version 1.4.2`. This will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes. If you're releasing a new major or minor version, this will create a new `docs/admin/release_notes/version_{major}.{minor}.md` file. You'll need to update the `Release Overview` section in that file manually and `git add mkdocs.yml pyproject.toml` to stage the changes made to the mkdocs navigation and the towncrier filename configuration.
```

> Generate release notes with `invoke generate-release-notes --version 1.4.2`. This will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes. If you're releasing a new major or minor version, this will create a new `docs/admin/release_notes/version_{major}.{minor}.md` file. You'll need to update the `Release Overview` section in that file manually and `git add mkdocs.yml pyproject.toml` to stage the changes made to the mkdocs navigation and the towncrier filename configuration.

Reviewer 2: As per the SR discussion, since we bump the version earlier using `poetry version type` and the script knows to look for it, I'd edit this to make `invoke generate-release-notes` the way we tell people to run it.

```suggestion
Generate release notes with `invoke generate-release-notes`. If you're releasing a new major or minor version, this will create a new `docs/admin/release_notes/version_{major}.{minor}.md` file. You'll need to update the `Release Overview` section in that file manually and `git add mkdocs.yml pyproject.toml` to stage the changes made to the mkdocs navigation and the towncrier filename configuration.

!!! note
    For a new release of `1.4.2`, this will update the release notes in `docs/admin/release_notes/version_1.4.md`, stage that file in git, and `git rm` all the fragments that have now been incorporated into the release notes.
    You can also use the `--version 1.4.2` parameter if you need to specify a version that's different from `pyproject.toml`.
```
